src/lib/BoundedCache.js

```diff
@@ -0,0 +1,62 @@
+import Cache from './Cache';
+
+class BoundedCache extends Cache {
+    /** @property {Array} - Maintains the list of cache keys in order in which they were added to the cache */
+    cacheQueue;
+
+    /** @property {number} - The maximum number of entries in the cache */
+    maxEntries;
+
+    /**
+     * [constructor]
+     *
+     * @param {number} [maxEntries] - Override the maximum number of cache entries
+     */
+    constructor(maxEntries) {
+        super();
+
+        this.maxEntries = maxEntries || 500;
+        this.cache = {};
+        this.cacheQueue = [];
```

Comment:
You could initialize `cacheQueue` as a class field, above the constructor.

No need to set `this.cache` since that will happen in the `super` call

Also, give a default value to `maxEntries` parameter, then you can remove the `||`